api/src/convert/compressed_ristretto.rs

```diff
@@ -23,7 +23,8 @@ impl TryFrom<&external::CompressedRistretto> for CompressedCommitment {
         if bytes.len() != 32 {
             return Err(ConversionError::ArrayCastError);
         }
-        let point = CompressedRistretto::from_slice(bytes);
+        let point = CompressedRistretto::from_slice(bytes)
+            .map_err(|_e| ConversionError::InvalidSignature)?;
```

Comment:
Is invalid signature the right error here?

Reply:
I'm not 100% sure given there's no documentation on the `ConversionError` variants. However, `ConversionError::InvalidSignature` is only used by the `archive_block` tests when an "ArchiveBlock with invalid metadata cannot be converted back to BlockData." So it seems that the "signature" in `InvalidSignature` refers to an invalid _block_ signature, and so it would be out of place here.

The `CompressedRistretto::from_slice` documentation states: 
```
    /// Returns [`TryFromSliceError`] if the input `bytes` slice does not have
    /// a length of 32.
```
So it seems that we should return a `ConversionError::ArrayCastError`, and we can remove the check on line 23 since the conversion already performs this length check.

Reply:
Done.